### PR TITLE
Specify primary datacenter for root cert watch

### LINKED ISSUE #361: Controller certificate watch fails in secondary datacenter
### Overview of the Issue

This issue is a contributing factor to https://github.com/hashicorp/consul-api-gateway/issues/300.

The controller fails to initialize a watch on root certificates when starting up in a secondary datacenter. This is because the primary datacenter needs to be included when creating the watch but is not today.

https://github.com/hashicorp/consul-api-gateway/blob/c35994bcb8fbcec332040acd1b7c6bb1d72e9d3e/internal/consul/certmanager.go#L215-L217

### Reproduction Steps

1. Create a federated setup by following the guide [Federation Between Kubernetes Clusters](https://www.consul.io/docs/k8s/deployment-configurations/multi-cluster/kubernetes). When installing, ensure API Gateway is enabled in both the primary and secondary clusters (`apiGateway.enabled: true` in values.yaml).
2. View logs of the API gateway controller in the secondary datacenter (`kubectl logs ...`)

### Logs

<details>
  <summary>Logs</summary>

```log
2022-09-06T19:58:26.971Z [INFO]  k8s/logger.go:30: consul-api-gateway-server.controller-runtime: Starting workers: controller=gateway controllerGroup=gateway.networking.k8s.io controllerKind=Gateway info="Starting workers" worker count=1
2022-09-06T19:58:26.971Z [INFO]  k8s/logger.go:30: consul-api-gateway-server.controller-runtime: Starting workers: controller=tcproute controllerGroup=gateway.networking.k8s.io controllerKind=TCPRoute info="Starting workers" worker count=1
2022-09-06T20:08:30.888Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:18:51.442Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:39:13.030Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:49:28.020Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:59:41.606Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T21:40:44.984Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T21:51:03.780Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
```

</details>

### Expected behavior

Controller successfully starts up (including cert watch initiated as part of startup)

### Environment details

If not already included, please provide the following:
- consul-api-gateway version: v0.4.0
- configuration used to deploy the gateway controller:
```yaml
apiGateway:
  enabled: true
  image: "hashicorp/consul-api-gateway:0.4.0"
```

Additionally, please provide details regarding the Kubernetes Infrastructure, as shown below:
- Consul Server version: v1.13.1
- Consul-K8s version: 0.47.1
- Cloud Provider (If self-hosted, the Kubernetes provider utilized): GKE

### PR DESCRIPTION
**Pull request by nathancoleman**
Fixes #361 

### Changes proposed in this PR:
In order to successfully watch root certs from a secondary datacenter, we must specify the name of the primary datacenter when setting up the watch. This requires that we accept the primary datacenter name as a flag when starting the controller. The other changes in this PR are just to plumb that value down into the watch initialization.

### How I've tested this PR:
1. Federate between 2 Kubernetes clusters as described in [this guide](https://www.consul.io/docs/k8s/deployment-configurations/multi-cluster/kubernetes), making sure to enable API gateway in both the primary and secondary clusters
    1. `apiGateway.enabled: true`
    2. `apiGateway.image: hashicorp/consul-api-gateway:0.4.0`
    3. `global.imageK8S: hashicorpdev/consul-k8s-control-plane:f1a9304a` (build from https://github.com/hashicorp/consul-k8s/pull/1481)
2. Create a `Gateway` in the secondary cluster and apply routes targeting services in the same cluster (I use the stack from [this Learn guide](https://learn.hashicorp.com/tutorials/consul/kubernetes-api-gateway))
3. Verify that you can reach these services. For the Learn guide, grab the IP from the `Gateway` status and visit `https://<ip>:8443`

<details>
<summary>Results from completing the above</summary>

```shell
➜  local git:(main) ✗ # In the secondary federated cluster
➜  local git:(main) ✗ kubectl config current-context
dc2
➜  local git:(main) ✗
➜  local git:(main) ✗ # Successfully deployed gateway
➜  local git:(main) ✗ k get gateway api-gateway -o yaml | yq .status
addresses:
  - type: IPAddress
    value: 34.66.212.71
conditions:
  - lastTransitionTime: "2022-09-15T19:36:12Z"
    message: Ready
    observedGeneration: 1
    reason: Ready
    status: "True"
    type: Ready
  - lastTransitionTime: "2022-09-15T19:36:12Z"
    message: Scheduled
    observedGeneration: 1
    reason: Scheduled
    status: "True"
    type: Scheduled
  - lastTransitionTime: "2022-09-15T19:36:12Z"
    message: InSync
    observedGeneration: 1
    reason: InSync
    status: "True"
    type: InSync
listeners:
  - attachedRoutes: 2
    conditions:
      - lastTransitionTime: "2022-09-15T19:44:01Z"
        message: NoConflicts
        observedGeneration: 1
        reason: NoConflicts
        status: "False"
        type: Conflicted
      - lastTransitionTime: "2022-09-15T19:44:01Z"
        message: Attached
        observedGeneration: 1
        reason: Attached
        status: "False"
        type: Detached
      - lastTransitionTime: "2022-09-15T19:44:01Z"
        message: Ready
        observedGeneration: 1
        reason: Ready
        status: "True"
        type: Ready
      - lastTransitionTime: "2022-09-15T19:44:01Z"
        message: ResolvedRefs
        observedGeneration: 1
        reason: ResolvedRefs
        status: "True"
        type: ResolvedRefs
    name: https
    supportedKinds:
      - group: gateway.networking.k8s.io
        kind: HTTPRoute
➜  local git:(main) ✗
➜  local git:(main) ✗ # Can reach hashicups through the gateway
➜  local git:(main) ✗ curl https://34.66.212.71:8443 --insecure
<!DOCTYPE html><html><head><meta name="viewport" content="width=device-width"/><meta charSet="utf-8"/><script>!function(){try {var d=document.documentElement.classList;d.remove('dark','light','dark','light');var e=localStorage.getItem('theme');if("system"===e||(!e&&true)){var t="(prefers-color-scheme: dark)",m=window.matchMedia(t);m.media!==t||m.matches?d.add('dark'):d.add('light')}else if(e) var x={"dark":"dark","light":"light","autoDark":"dark","autoLight":"light"};d.add(x[e])}catch(e){}}()</script><title>HashiCups - Demo App</title>...
```

</details>

### How I expect reviewers to test this PR:
See above

### Checklist:
- [ ] Tests added
- [x] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: internal/consul/certmanager.go [hunk 1]
```diff
@@ -213,7 +216,8 @@ func (c *CertManager) Manage(ctx context.Context) error {
 	c.logger.Trace("running cert manager")
 
 	rootWatch, err := watch.Parse(map[string]interface{}{
-		"type": "connect_roots",
+		"datacenter": c.primaryDatacenter,
```
Comment (nathancoleman) [inline internal/consul/certmanager.go:219]: Every other change in this PR is to facilitate this one line where we specify the primary datacenter for our root cert watch. This is necessary for Gateways to work in the secondary datacenter of a federated setup.

Comment (mikemorris) [reply internal/consul/certmanager.go:219]: When this flag in not set, does defaulting to empty string for this field infer the current datacenter and match existing behavior?

**Review by mikemorris — APPROVED** [review]
LGTM pending question about default behavior when the `-primary-datacenter` flag isn't set and fixing the Consul Enterprise license to get CI tests passing.

**Comment by nathancoleman** [comment]
@mikemorris thoughts on the release note added here?

**Pull request merged**